beacon_chain/networking/eth2_network.nim

```diff
@@ -1290,7 +1297,7 @@ proc checkPeer(node: Eth2Node, peerAddr: PeerAddr): bool =
     else:
       true
 
-proc dialPeer(node: Eth2Node, peerAddr: PeerAddr, index = 0) {.async.} =
+proc dialPeer(node: Eth2Node, peerAddr: PeerAddr, index = 0, isnotTor = true ) {.async.} =
```

Comment:
Basically same comment as the `broadcastTor` and `broadcast` coexisting: this seems like an auditing challenge, to make sure that a function with a security-important optional parameter is always called correctly.

Reply:
Maybe we can do something like that:

```
proc dialPeer(node: Eth2Node, peerAddr: PeerAddr, index = 0) {.async.} =
  await internalDialPeer(node, peerAddr, node.switch)

proc dialPeerThroughTor(node: Eth2Node, peerAddr: PeerAddr, index = 0) {.async.} =
  await internalDialPeer(node, peerAddr, node.torSwitch)
```

Reply:
Okay that can be separately called. That would be part of other refactoring suggested.